CMakeLists.txt

```diff
@@ -93,18 +93,31 @@ else()
   if(ENABLE_CUBLAS_BACKEND OR ENABLE_CURAND_BACKEND)
     set(CMAKE_CXX_COMPILER "clang++")
   elseif(ENABLE_MKLGPU_BACKEND)
-    set(CMAKE_CXX_COMPILER "dpcpp")
+    if(UNIX)
+      set(CMAKE_CXX_COMPILER "icpx")
+    else()
+      set(CMAKE_CXX_COMPILER "icx")
+    endif()
   else()
-    find_program(DPCPP_PATH dpcpp)
-    if(DPCPP_PATH)
-      message(STATUS "CXX compiler: dpcpp was found in PATH, using dpcpp")
-      set(CMAKE_CXX_COMPILER "dpcpp")
+    if(UNIX)
+      find_program(ICPX_ICX_PATH icpx)
+    else()
+      find_program(ICPX_ICX_PATH icx)
+    endif()
+    if(ICPX_ICX_PATH)
+      if(UNIX)
+        message(STATUS "CXX compiler: icpx was found in PATH, using icpx")
+        set(CMAKE_CXX_COMPILER "icpx")
+      else()
+        message(STATUS "CXX compiler: icx was found in PATH, using icx")
+        set(CMAKE_CXX_COMPILER "icx")
+      endif()
     else()
       if(WIN32)
-        message(STATUS "CXX compiler: dpcpp was not found in PATH, using clang-cl instead")
+        message(STATUS "CXX compiler: icpx/icx was not found in PATH, using clang-cl instead")
```

Comment:
I think we could put `icx` here to be more specific.

Reply:
Thank you! Changed it to icx